navigation-experimental/src/main/kotlin/com/freeletics/khonshu/navigation/internal/MultiStackNavigationExecutorBuilder.kt

```diff
@@ -28,16 +28,28 @@ internal fun rememberNavigationExecutor(
     deepLinkPrefixes: Set<DeepLinkHandler.Prefix>,
 ): MultiStackNavigationExecutor {
     val context = LocalContext.current
+
     val viewModel = viewModel<StoreViewModel>(factory = SavedStateViewModelFactory())
+    viewModel.setInputStartRoot(startRoot)
+
     return remember(context, viewModel) {
         val contentDestinations = destinations.filterIsInstance<ContentDestination<*>>()
         val activityDestinations = destinations.filterIsInstance<ActivityDestination>()
 
         val navState = viewModel.globalSavedStateHandle.get<Bundle>(SAVED_STATE_STACK)
         val stack = if (navState == null) {
-            MultiStack.createWith(startRoot, contentDestinations, viewModel::removeEntry)
+            MultiStack.createWith(
+                root = viewModel.savedNavRoot!!,
+                destinations = contentDestinations,
+                onStackEntryRemoved = viewModel::removeEntry,
+            )
         } else {
-            MultiStack.fromState(startRoot, navState, contentDestinations, viewModel::removeEntry)
+            MultiStack.fromState(
+                root = viewModel.savedNavRoot!!,
+                bundle = navState,
+                destinations = contentDestinations,
+                onStackEntryRemoved = viewModel::removeEntry,
+            )
```

Comment:
Instead of saving the startRoot through the `ViewModel` we could also just save it as part of `MultiStack` itself and remove the `root` parameter from `fromState`. That would ensure consistency and would be less complex.

Reply:
Done https://github.com/hoc081098/khonshu/compare/25c159adacb0194d510772c9989d425107ecaac2...hoc081098:khonshu:82e02b51b28fe35c286b9bcae33553e0628e4d72 🙏